src/WrappedWorker.js

```diff
@@ -0,0 +1,221 @@
+"use strict";
+
+var _ = require('underscore');
+var Backbone = require('backbone');
+
+function makeDeferred() {
+    var dfd = {};
+    dfd.promise = new Promise(function (resolve, reject) {
+        dfd.resolve = resolve;
+        dfd.reject = reject;
+    });
+    return dfd;
+}
+
+/**
+ * Wrapper around the global "Worker" thread constructor
+ * that only allows one message to the worker at a time.
+ * Will wait until previous message has received a 
+ * "response" before posting a mew message.
+ * 
+ * Example usage:
+ * var worker = new WrappedWorker({ workerFilePath: 'path/to/file' });
+ * worker.send({ someMessage: 'foo' })
+ *      .then(function (event) {
+ *          console.log('response from worker', event.data);
+ *      });
+ *
+ * 
+ * @class WrappedWorker
+ * @param {object} options
+ * @param {string} options.workerFilePath
+ *      The path to the file that should be loaded when the 
+ *      worker is created
+ * @param {string} [options.Worker]
+ *      The worker constructor you want to use. Defaults
+ *      to window.Worker. This is primarily intended for
+ *      testing.
+ */
+function WrappedWorker(options) {
+    this.initialize(options);
+}
+
+// accessible for testing
+if (typeof Worker !== "undefined") {
+    WrappedWorker._Worker = Worker;
+}
+
+// make WrappedWorker extend-able
+WrappedWorker.extend = Backbone.Model.extend;
```

Comment:
Sure, makes sense. I can use the [simple-extend](https://www.npmjs.com/package/simple-extend) library, if you prefer. I wanted people to be able to extend and supply their own wrapped worker system to manage workers, so that is why I made it extendable.